### PR TITLE
search frontend: fix predicate highlights for field aliases

### LINKED ISSUE #21880: Predicates broken when using filter alias
Works: [`repo:contains.file(CHANGELOG)`](https://sourcegraph.com/search?q=context:global+repo:contains.file%28CHANGELOG%29&patternType=literal)

Borked: [`r:contains.file(CHANGELOG)`](https://sourcegraph.com/search?q=context:global+r:contains.file%28CHANGELOG%29&patternType=literal)

When using the `r:` alias, predicates are present in suggestions, but predicate highlighting does not work -- the filter value is highlighted as regexp:

image

image

### PR DESCRIPTION
**Pull request by rvantonder**
Fixes #21880.

Screen Shot 2021-06-15 at 3 36 42 PM

### CODE REVIEW

File: client/shared/src/search/query/filters.ts [hunk 1]
```diff
@@ -49,6 +49,27 @@ export enum AliasedFilterType {
 }
 /* eslint-enable unicorn/prevent-abbreviations */
 
+export const ALIASES: Record<string, string> = {
```
Comment (rvantonder) [inline client/shared/src/search/query/filters.ts:52]: We have this other alias enum type `AliasedFilterType `, and a helper function, to resolve aliases but the type signatures are really awkward to use. I will refactor things to use this simpler record data type instead later.

File: client/shared/src/search/query/filters.ts [hunk 2]
```diff
@@ -49,6 +49,27 @@ export enum AliasedFilterType {
...
+    since: 'after',
+    until: 'before',
+    m: 'message',
+    msg: 'message',
+    revision: 'rev',
+}
+
+export const resolveFieldAlias = (field: string): string => {
+    const canonical = ALIASES[field]
+    if (canonical !== undefined) {
```
Comment (camdencheek) [inline client/shared/src/search/query/filters.ts:67]: Forgive my {java,type}script noobness, but what's the difference between this and `if (canonical) {`?

**Review by camdencheek — APPROVED** [review]

File: client/shared/src/search/query/filters.ts (hunk 2 above)
Comment (rvantonder) [reply client/shared/src/search/query/filters.ts:67]: my noobness

Comment (rvantonder) [reply client/shared/src/search/query/filters.ts]: `export const resolveFieldAlias = (field: string): string => ALIASES[field] || field`

Comment (camdencheek) [reply client/shared/src/search/query/filters.ts]: go -> typescript transliteration strikes again

**Pull request merged**